fitlins/interfaces/afni.py

```diff
@@ -0,0 +1,637 @@
+            to be included in the regression model of the respective run_img.
+            The number of rows must match the number of volumes in the
+            respective run_img. Ignored in case designs is not None.
+            If string, then a path to a csv file is expected.
+
+        design_matrices : pandas DataFrame or list of pandas DataFrames,
+            Design matrices that will be used to fit the GLM. If given it
+            takes precedence over events and confounds.
+
+        """
+        import sys
+        import time
+        from nilearn._utils.niimg_conversions import check_niimg
+
+        from nistats.design_matrix import make_first_level_design_matrix
+        from nistats.utils import (
+            _check_run_tables,
+            _check_events_file_uses_tab_separators,
+        )
+
+        # Check arguments
+        # Check imgs type
+        if events is not None:
+            _check_events_file_uses_tab_separators(events_files=events)
+        if not isinstance(run_imgs, (list, tuple)):
+            run_imgs = [run_imgs]
+        if design_matrices is None:
+            if events is None:
+                raise ValueError("events or design matrices must be provided")
+            if nistats_flm.t_r is None:
+                raise ValueError(
+                    "t_r not given to FirstLevelModel object"
+                    " to compute design from events"
+                )
+        else:
+            design_matrices = _check_run_tables(
+                run_imgs, design_matrices, "design_matrices"
+            )
+        # Check that number of events and confound files match number of runs
+        # Also check that events and confound files can be loaded as DataFrame
+        if events is not None:
+            events = _check_run_tables(run_imgs, events, "events")
+        if confounds is not None:
+            confounds = _check_run_tables(run_imgs, confounds, "confounds")
+
+        # For each run fit the model and keep only the regression results.
+        nistats_flm.labels_, nistats_flm.results_, self.design_matrices_ = ([], [], [])
+        n_runs = len(run_imgs)
+        t0 = time.time()
+        for run_idx, run_img in enumerate(run_imgs):
+            # Report progress
+            if nistats_flm.verbose > 0:
+                percent = float(run_idx) / n_runs
+                percent = round(percent * 100, 2)
+                dt = time.time() - t0
+                # We use a max to avoid a division by zero
+                if run_idx == 0:
+                    remaining = "go take a coffee, a big one"
+                else:
+                    remaining = (100.0 - percent) / max(0.01, percent) * dt
+                    remaining = "%i seconds remaining" % remaining
+
+                sys.stderr.write(
+                    "Computing run %d out of %d runs (%s)\n"
+                    % (run_idx + 1, n_runs, remaining)
+                )
```

Comment:
For messages like this, we can use logging, such as:

```Python
from nipype import logging
logger = logging.getLogger('nipype.interface')

logger.debug(f"Computing run {run_idx + 1} out of {n_runs} runs ({remaining})")

Reply:
noted